docs/lsp.md

```diff
@@ -0,0 +1,10 @@
+# Setting up LSP with clangd
+## Background
+Editors such as Emacs, Vim, or VS Code support features like code navigation, code completion, inline compiler error messages, and others, through [LSP](https://en.wikipedia.org/wiki/Language_Server_Protocol), the Language Server Protocol. A common language server with LSP support is [clangd](https://clangd.llvm.org), which relies on the presence of `compile_commands.json`, a JSON file with a record of the compile commands for each file in a project.
+## How do I generate `compile_commands.json` for XLA source code?
+Use the [build_tools/lint/generate_compile_commands.py](https://github.com/openxla/xla/blob/main/build_tools/lint/generate_compile_commands.py) script. At the time of writing, the following invocation from XLA repo root generates a `compile_commands.json` file in place:
```

Comment:
Probably not a blocker for this CL, but instead of saying "at the time of writing", surely we could just add a test to CI?

Reply:
A test is a good idea, looks like @ddunl is working on one. The "at the time of writing" is defensive documentation in case a test like that isn't enabled or doesn't exist in the first place :) Should I remove that phrasing for simplicity?

Reply:
maybe let's remove/hope the test gets done

Reply:
Removed defensive phrasing.